### PR TITLE
fix: Moved _Setting up your environment_ to shared repo.

### PR DESCRIPTION
**Pull request by stoobie**
### Description of the Changes

Moved _Setting up your environment_ to shared repo, and edited it for clarity.

### PR Preview URL

https://starknet-io.github.io/starknet-docs/pr-847/documentation/quick_start/environment_setup/

### Check List

- [x] Changes have been done against dev branch, and PR does not conflict
- [x] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/847)

### CODE REVIEW

**Review by JameStark — COMMENTED** [review]
Screenshot 2023-09-19 at 10 22 33

The gap here looks too big

File: components/Starknet/modules/quick_start/pages/environment_setup.adoc [hunk 1]
```diff
@@ -1,101 +1 @@
...
-----
-
-You can run the same command to upgrade Scarb:
-
-[source, bash]
-----
-curl --proto '=https' --tlsv1.2 -sSf https://docs.swmansion.com/scarb/install.sh | sh
-----
\ No newline at end of file
+include::1.0.0@docs-common-content:ROOT:partial$partial_environment_setup.adoc[]
```
Comment (stoobie) [inline components/Starknet/modules/quick_start/pages/environment_setup.adoc:1]:
```suggestion
include::1.0.0@docs-common-content:ROOT:partial$partial_environment_setup.adoc[]

```

**Comment by stoobie** [comment]
> Screenshot 2023-09-19 at 10 22 33
> The gap here looks too big

Fixed

**Comment by JameStark** [comment]
Content aside, this first section still doesn't look right visually to me 
Screenshot 2023-09-27 at 12 10 32

vs what we currently have

Screenshot 2023-09-27 at 12 10 43

File: playbook.yml [hunk 2]
```diff
@@ -10,12 +10,14 @@ urls:
...
-      branches: HEAD
+      branches:
+        - 'HEAD'
       edit_url: 'https://github.com/starknet-io/starknet-docs/edit/dev/{path}'
       start_paths:
         - components/Starknet
     - url: https://github.com/starknet-io/docs-common-content.git
-      branches: HEAD
+      branches: 'steve/Fix_intro'
+#      branches: 'HEAD'
```
Comment (stoobie) [inline playbook.yml:20]:
```suggestion
      branches: 'HEAD'
```

File: components/Starknet/modules/quick_start/nav.adoc [hunk 3]
```diff
@@ -1,8 +1,9 @@
 * Quick start
 
 ** xref:environment_setup.adoc[Setting up your environment]
-*** xref:environment_setup.adoc#installing_starkli[Installing Starkli]
-*** xref:environment_setup.adoc#installing_scarb[Installing Scarb]
+*** xref:environment_setup.adoc#installing_starkli[Starkli installation]
```
Comment (stoobie) [inline components/Starknet/modules/quick_start/nav.adoc:4]:
```suggestion
*** xref:environment_setup.adoc#installing_starkli[Installing Starkli]
```

File: components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc [hunk 4]
```diff
@@ -21,9 +21,9 @@ In Starknet, similar to Ethereum, every contract has a nonce. This nonce is sequ
 
 [NOTE]
 ====
-In Starknet, only the nonce of account contracts, that is, those adhering to the above structure, can be non-zero. 
+In Starknet, only the nonce of account contracts, that is, those adhering to the above structure, can be non-zero.
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc:24]:
```suggestion
In Starknet, only the nonce of account contracts, that is, those adhering to the above structure, can be non-zero. 
```

File: components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc [hunk 5]
```diff
@@ -21,9 +21,9 @@ In Starknet, similar to Ethereum, every contract has a nonce. This nonce is sequ
...
 
-In contrast, in Ethereum, regular smart contracts, known as _Contract Accounts_, as opposed to _Externally Owned Accounts_ can increment their nonce by deploying smart contracts, that is, executing the `CREATE` and `CREATE2` opcodes. 
+In contrast, in Ethereum, regular smart contracts, known as _Contract Accounts_, as opposed to _Externally Owned Accounts_ can increment their nonce by deploying smart contracts, that is, executing the `CREATE` and `CREATE2` opcodes.
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc:26]:
```suggestion
In contrast, in Ethereum, regular smart contracts, known as _Contract Accounts_, as opposed to _Externally Owned Accounts_ can increment their nonce by deploying smart contracts, that is, executing the `CREATE` and `CREATE2` opcodes. 
```

File: components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc [hunk 6]
```diff
@@ -105,7 +105,7 @@ You can find the compiler version supported by Starkli by running:
 
 [source,bash]
 ----
-starkli declare --help 
+starkli declare --help
```
Comment (stoobie) [inline components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc:108]:
```suggestion
starkli declare --help 
```

File: components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc [hunk 7]
```diff
@@ -139,7 +139,7 @@ If you were using a provider like Infura or Alchemy, the declaration command wou
 [source,bash]
 ----
 starkli declare target/dev/contracts_Ownable.sierra.json \
-    --rpc=https://starknet-goerli.infura.io/v3/<API_KEY> \ 
+    --rpc=https://starknet-goerli.infura.io/v3/<API_KEY> \
```
Comment (stoobie) [inline components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc:142]:
```suggestion
    --rpc=https://starknet-goerli.infura.io/v3/<API_KEY> \ 
```

**Pull request merged**